lsp-metals.el

```diff
@@ -393,37 +409,47 @@ change it again."
     (erase-buffer)
     (insert value)
     (goto-char (point-min))
+    (when (string-suffix-p "javap" uri)
```

Comment:
I don't think we'd want to activate `java-mode` in the case of `javap-verbose` since the extra info it returns isn't valid java or at least doesn't look nice under java mode fontification (see `javap -v java.lang.Object`), so the predicate succeeds for javap only if that makes sense.

Reply:
ok, then maybe just `string=` should be used here?

Reply:
`uri` is (typically) of the form `metalsDecode:file:///somePath/someFile.scala.javap` (see https://scalameta.org/metals/docs/integrations/new-editor/#decode-file) rather than a format id like `javap`.

I could forward `format-id` to `lsp-metals--decode` from `lsp-metals-decode-file` if you think an equality check would be better?

Reply:
Ah, sorry. I missed the fact that it matches url but not format-id. Then it's fine.